### PR TITLE
provider/archive support folders in output_path

### PR DESCRIPTION
**Pull request by nicolai86**
The `archive` provider right now relies on `FileInfo`s `Name()` function, which only returns the base name of the`output_path`, ignoring any folders:

```
type FileInfo interface {
        Name() string       // base name of the file
        Size() int64        // length in bytes for regular files; system-dependent for others
        Mode() FileMode     // file mode bits
        ModTime() time.Time // modification time
        IsDir() bool        // abbreviation for Mode().IsDir()
        Sys() interface{}   // underlying data source (can return nil)
}
```

this PR changes to use the `output_path` directly, allowing archives to be placed in subfolders:

before:

```
resource "archive_file" "foo" {
  type                    = "zip"
  source_content          = "This is some content"
  source_content_filename = "content.txt"
  output_path             = "example/path/test.zip"
}
```

throws error that `test.zip` could not be found

after: creates test.zip in `example/path`.

### CODE REVIEW

File: builtin/providers/archive/resource_archive_file.go [hunk 1]
```diff
@@ -95,9 +97,18 @@ func resourceArchiveFileRead(d *schema.ResourceData, meta interface{}) error {
 
 func resourceArchiveFileUpdate(d *schema.ResourceData, meta interface{}) error {
 	archiveType := d.Get("type").(string)
-	outputPath := d.Get("output_path").(string)
+	output_path := d.Get("output_path").(string)
```
Comment (kwilczynski) [inline builtin/providers/archive/resource_archive_file.go]: I am not in favour of this. The camel case is the preferred in Go.

**Comment by nicolai86** [comment]
@kwilczynski agreed. The file used both styles and I wanted a unique way to name the same thing. i can adjust it

**Comment by kwilczynski** [comment]
@nicolai86 definitely something to fix, go for it :)

**Comment by nicolai86** [comment]
@kwilczynski I've adjusted the variable name now.

**Comment by stack72** [comment]
Hi @nicolai86 

The PR looks good to me now :) The tests are green as expected:

```
% make testacc TEST=./builtin/providers/archive TESTARGS='-run=Test'                              1 ↵
==> Checking that code complies with gofmt requirements...
/Users/stacko/Code/go/bin/stringer
go generate $(go list ./... | grep -v /terraform/vendor/)
2016/08/18 16:33:50 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/archive -v -run=Test -timeout 120m
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestAccArchiveFile_Basic
--- PASS: TestAccArchiveFile_Basic (0.07s)
=== RUN   TestZipArchiver_Content
--- PASS: TestZipArchiver_Content (0.00s)
=== RUN   TestZipArchiver_File
--- PASS: TestZipArchiver_File (0.00s)
=== RUN   TestZipArchiver_Dir
--- PASS: TestZipArchiver_Dir (0.00s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/archive	0.093s
```

Paul

**Comment by nicolai86** [comment]
🎉

**Comment by nicolai86** [comment]
@stack72 told me he gets permission errors in the example folder since the merge. I'll take a look at this today evening

**Comment by stack72** [comment]
thanks @nicolai86 

After running the archive tests, when i try and build or run any other tests, i get the following:

```
% make plugin-dev PLUGIN=provider-aws                                                             1 ↵
/Users/stacko/Code/go/bin/stringer
go generate $(go list ./... | grep -v /terraform/vendor/)
open builtin/providers/archive/example: permission denied
2016/08/18 16:47:28 Failed to discover providers: Failed parsing directory builtin/providers/archive/example: open builtin/providers/archive/example: permission denied
exit status 1
config.go:1: running "go": exit status 1
make: *** [generate] Error 1
```

**Comment by nicolai86** [comment]
Okay, the issue is that I'm not giving `u+r` permission for the test folder.
I'll solve this issue in two ways: a) correct permissions b) use a tmp folder for the output path, this way I also do not pollute the terraform source folders… PR incoming

**Comment by nicolai86** [comment]
I've created a new PR to address the test issue. sorry for the trouble -.-

**Comment by stack72** [comment]
You rock @nicolai86 :) Thanks!

**Pull request merged**